odml/templates.py

```diff
@@ -0,0 +1,188 @@
+"""
+Handles (deferred) loading of odML templates
+"""
+
+import os
+import sys
+import tempfile
+import threading
+try:
+    import urllib.request as urllib2
+    from urllib.error import URLError
+    from urllib.parse import urljoin
+
+except ImportError:
+    import urllib2
+    from urllib2 import URLError
+    from urlparse import urljoin
+
+from datetime import datetime as dati
+from datetime import timedelta
+from hashlib import md5
+
+from .tools.parser_utils import ParserException
+from .tools.xmlparser import XMLReader
+
+
+REPOSITORY_BASE = 'https://templates.g-node.org/'
+REPOSITORY = urljoin(REPOSITORY_BASE, 'templates.xml')
+
+CACHE_AGE = timedelta(days=1)
+CACHE_DIR = "odml.cache"
+
+
+# TODO after prototyping move functions common with
+# terminologies to a common file.
+
+
+def cache_load(url):
+    """
+    Load the url and store the file in a temporary cache directory.
+    Subsequent requests for this url will use the cached version until
+    the file is older than the CACHE_AGE.
+
+    Exceptions are caught and not re-raised to enable loading of nested
+    odML files without breaking if one of the child files is unavailable.
+
+    :param url: location of an odML template XML file.
+    :return: Local file location of the requested file.
+    """
+
+    filename = '.'.join([md5(url.encode()).hexdigest(), os.path.basename(url)])
+    cache_dir = os.path.join(tempfile.gettempdir(), CACHE_DIR)
+
+    # Create temporary folder if required
+    if not os.path.exists(cache_dir):
+        try:
+            os.makedirs(cache_dir)
+        except OSError:  # might happen due to concurrency
+            if not os.path.exists(cache_dir):
+                raise
```

Comment:
Using the `exist_ok=True` argument to `os.makedirs()` should accomplish the same purpose as this block, I think.

Reply:
We'll keep it in mind if we no longer support python 2, the `exist_ok` is python 3 only.